### ISSUE TITLE
[java] JUnit4TestShouldUseTestAnnotation false positive not in test code

### ISSUE THREAD

**Issue by mkordas** [pmd/pmd#944]
**Affects PMD Version:** 
6.0.1

**Rule:**
JUnit4TestShouldUseTestAnnotation

**Description:**
JUnit4TestShouldUseTestAnnotation triggers in production code, when method named `test()` is used (e.g.) to implement predicate. This rule should not trigger for classes that don't have `Test` suffix.

**Code Sample demonstrating the issue:**

```java
public final class AlwaysExists implements Predicate {

	@Override
	public boolean test(String value) { // [JUnit4TestShouldUseTestAnnotation] JUnit 4 tests that execute tests should use the @Test annotation
		return true;
	}
}
```

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
Marking as duplicate of the older #572, will follow up there.
